### ISSUE TITLE
Formatting code with vector macro and >= two impl methods causes compiler error

### ISSUE THREAD

**Issue by AnthonyMansour** [rust-lang/rls-vscode#617]
Formatting the following code causes a needed closing curly bracket `}` to be deleted and so a compiler error occurs.

**Before formatting:**
```
struct FormatTest {format_test: Vec<Option<i32>>,}

impl FormatTest {
    fn test1() -> i32{
        return 0;
        }
    
    fn test2(test_val: usize) -> Self{FormatTest {format_test: vec![None; test_val]}}
}


fn main() {
    FormatTest::test2(5);
}
```

**After Formatting (does not compile):**
```
struct FormatTest {
    format_test: Vec<Option<i32>>,
}

impl FormatTest {
    fn test1() -> i32 {
        return 0;
    }

    fn test2(test_val: usize) -> Self {
        FormatTest {
            format_test: vec![None; test_val],
    }
}


fn main() {
    FormatTest::test2(5);
}
```

**Compiler error:**
```
this file contains an un-closed delimiter
main.rs(5, 17): un-closed delimiter
main.rs(11, 20): this delimiter might not be properly closed...
main.rs(13, 5): ...as it matches this but it has different indentation
```

**Comment by Xanewok** [comment]
This looks like an error that's related to the Rust 1.35 release. Could you try 1.36 beta (or wait till Thursday for it to released as stable) to see if the issue persists there?

cc https://github.com/rust-lang/rls-vscode/issues/576

**Comment by AnthonyMansour** [comment]
> This looks like an error that's related to the Rust 1.35 release. Could you try 1.36 beta (or wait till Thursday for it to released as stable) to see if the issue persists there?
> 
> cc #576

Issue fixed in Rust 1.36
